src/components/AdminForm.js

```diff
@@ -1,38 +1,91 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { connect } from "react-redux";
+import { Redirect } from "react-router-dom";
+
 import {
   setCohortName,
-  setCohortType
+  setCohortType,
+  postFormDetailsThunk,
+  setResetApp,
+  resetIsSubmitted
 } from "../redux/actions/adminFormActions";
+import { resetDates } from "../redux/actions/dateActions";
 
 import TextInput from "./TextInput";
 import Select from "./Select";
-import DatePickerContainer from "./DatePickerContainer";
 import SubmitButton from "./SubmitButton";
-import Questions from './Questions/Questions';
-
+import DatePickerContainer from "./DatePickerContainer";
+import Questions from "./Questions/Questions";
 
 const AdminForm = props => {
+  const [isDuplicate, setDuplicate] = useState(false);
+  console.log(props);
+  useEffect(() => {
+    props.setResetApp();
+    props.resetDates();
+  }, []);
+
   const handleCohortNameChange = e => {
     props.setCohortName(e.target.value);
   };
 
-  const handleCohortTypeChange = e => {
-    props.setCohortType(e.target.value);
+  /**
+   * Check if the cohort name and type already exists in databases
+   * If that already exists then the returned length will be greater than 0
+   */
+  const isCohortDuplicate = () => {
+    const { cohortName, cohortType, existingCohorts } = props;
+    return existingCohorts
+      .filter(cohort => cohort.cohortType === cohortType)
+      .filter(
+        cohort => cohort.cohortName.toLowerCase() === cohortName.toLowerCase()
+      ).length;
   };
 
   const handleSubmit = e => {
     e.preventDefault();
-    alert("Submitting");
-    // on submit we need to save the form in DB
+    const {
+      cohortName,
+      cohortType,
+      dateOpen,
+      dateClose,
+      dateOfResponse
+    } = props;
+
+    const isCohortDuplicateValue = isCohortDuplicate();
+    //if that recotrd doesnt exist in database then add it to database
+    if (isCohortDuplicateValue === 0) {
+      const cohortData = {
+        cohortName,
+        cohortType,
+        link: "/",
+        dateOpen,
+        dateClose,
+        dateOfResponse
+      };
+      //calls the thunk here to "POST" to database
+      props.postFormDetailsThunk(cohortData);
+    } else {
+      setDuplicate(true);
+    }
+  };
+
+  const handleCohortTypeChange = e => {
+    props.setCohortType(e.target.value);
   };
 
   const selectOptions = [
-    { value: "frontend", displayedName: "Frontend" },
-    { value: "backend", displayedName: "Backend" },
-    { value: "productDesign", displayedName: "Product Design" }
+    { value: "Frontend Development", displayedName: "Frontend Development" },
```

Comment:
We should bring the values in these option objects to what they were before, for example 'value: Frontend Development' should go back to 'frontend' or if needed we can do 'frontend-development'. This is needed for standardizing the values in the database so that we can easier filter them later.